### PR TITLE
Added extra playback keybindings

### PR DESCRIPTION
**Pull request by mijofa**
**Changes**
Added playback keybindings for:
* ArrowUp & ArrowDown: Volume up & down
* Shift-P & Shift-N: Previous & next track
* 0-9: Jump to 0,10,20,30...90% of the current media
* Home, & End: Jump to 0% or 100% of the current media

**Issues**
N/A

**Other info**
I specifically wanted the vol up/down bindings, so I'm happy if only those get merged in. I did the rest because I felt some sort of defined standard should be followed, so I reffered to [YouTube's bindings](https://support.google.com/youtube/answer/7631406?hl=en) and implemented all of them.

This was also discussed in the Matrix/IRC channel:
> 2020-02-19T20:21:54+1100 <mijofa_> Is there any intention of (or interest in) adding extra keybindings to the web video player?        I've had a snippet of js that I wrote for Emby I've been dropping on the end of videoosd.js with each update, but I'm considering making it into an actual patch and pushing it upstream if it's actually got a chance of being merged in
> 2020-02-19T20:22:35+1100 <dkanada[m]> whats added?
> 2020-02-19T20:24:15+1100 <mijofa_> Main things I wanted/needed was play/pause on space, seek on left/right, and volume on up/down. But I added all the ones YouTube uses as documented at https://support.google.com/youtube/answer/7631406
> 2020-02-19T20:25:15+1100 <mijofa_> http://ix.io/2c7E/js
> 2020-02-19T20:25:57+1100 <telans[m]>  I can seem to seek & pause with keys at the moment
> 2020-02-19T20:26:09+1100 <mijofa_> (disclaimer, that code is horrible, it was written in the least "intrusive" way possible)
> 2020-02-19T20:26:42+1100 <mijofa_> If I recall correctly my main issue with the play/pause was that it wasn't entirely consistent, you needed to be focussed on the right element and the OSD needed to not be active. Or something like that
> 2020-02-19T20:27:38+1100 <mijofa_> Also I'm sure that some of this would've gotten better since I actually wrote it, but it's still not quite good enough for me
> 2020-02-19T20:44:05+1100 <JustAMan[m]> mijofa_: make a PR, but check existing ones first
> 2020-02-19T20:44:13+1100 <JustAMan[m]> I think I saw something pretty similar recently
> 2020-02-19T20:44:33+1100 <mijofa_> JustAMan[m]: Ok, will do, thanks. :)
> 2020-02-19T20:44:50+1100 <dmitrylyzo[m]> Someone also wants Space as play/pause
> 2020-02-19T20:44:50+1100 <dmitrylyzo[m]> https://github.com/jellyfin/jellyfin-web/pull/805
> 2020-02-19T20:44:52+1100 <dkanada[m]> pretty sure that already works, save the minor bug you mentioned
> 2020-02-19T20:45:10+1100 <dkanada[m]> <dkanada[m] "pretty sure that already works, "> which has been resolved in the PR linked above
> 2020-02-19T20:47:05+1100 <MrTimscampi[m]> Addendum, for left/right seek, it needs to be disabled in TV layout, since navigation there uses the directional keys
> 2020-02-19T20:47:39+1100 <dmitrylyzo[m]> up/down also
> 2020-02-19T20:47:41+1100 <mijofa_> I would expect this to only affect the media player when it's currently in focus and playing the media
> 2020-02-19T20:48:40+1100 <mijofa_> And yeah, I see in the code where it's already doing left/right seeking

### CODE REVIEW

**Comment by jdyyc** [comment]
Aw damn.    Trying to figure out how to do my first ever git pull request and I came here to see that someone beat me to it by 5 hours!

My change uses the U and P keys for previous and next respectively as those are directly above the ff/rw keys  (J and L).  

One thing I did that you're missing is updating the tooltips so that when you mouseover the buttons it shows the function and their keyboard shortcuts.  Something like this in videoosd.html...

`<button is="paper-icon-button-light" class="btnPreviousTrack autoSize hide" title="${ButtonPreviousTrack} (u)">`

**Comment by mijofa** [comment]
> Aw damn. Trying to figure out how to do my first ever git pull request and I came here to see that someone beat me to it by 5 hours!

Hah! Sorry about that. xD

> My change uses the U and P keys for previous and next respectively as those are directly above the ff/rw keys (J and L).

Ah, yeah, well positioned on the keyboard, I was concerned picking keys like that would end up setting a precedent of not actually following standards. Not sure I chose a good standard, but choosing a bad one is better than not choosing one at all.

> One thing I did that you're missing is updating the tooltips so that when you mouseover the buttons it shows the function and their keyboard shortcuts. Something like this in videoosd.html...
> 
> `<button is="paper-icon-button-light" class="btnPreviousTrack autoSize hide" title="${ButtonPreviousTrack} (u)">`

That's not a bad idea. I reckon go ahead with that one, I'm not sure if you can add a commit to this pull request, but I'm more than happy for that to be included in this one. Hasn't had any review from the Jellyfin devs yet though/

**Comment by dmitrylyzo** [comment]
It seems to work (Desktop/TV[_except volume due to navigation_]).

Just thoughts - no need to solve immediately.
* OSD is not shown on binding trigger, as for existing bindings. But some do not like OSD while seeking. Some lightweight OSD like volumeosd may help.
* Volume changing spams server. Need something like `setVolume/setVolumeDelayed`.

**Comment by dkanada** [comment]
@dmitrylyzo the volume issue has been fixed by now right?

**Comment by dmitrylyzo** [comment]
> @dmitrylyzo the volume issue has been fixed by now right?

Volume change spams with `reportPlaybackProgress` on each `keydown` because `playbackManager.volumeUp`/`playbackManager.volumeDown` are used.

<details>
<summary>Stack</summary>

```
reportPlayback (playbackmanager.js?v=10.5.0:43)
sendProgressUpdate (playbackmanager.js?v=10.5.0:3390)
onPlaybackVolumeChange (playbackmanager.js?v=10.5.0:3281)
(anonymous) (events.js?v=10.5.0:26)
trigger (events.js?v=10.5.0:25)
onVolumeChange (plugin.js?v=10.5.0:830)
----------volumechange (async)----------
HtmlVideoPlayer.setVolume (plugin.js?v=10.5.0:1722)
HtmlVideoPlayer.volumeDown (plugin.js?v=10.5.0:1738)
PlaybackManager.self.volumeDown (playbackmanager.js?v=10.5.0:1252)
onWindowKeyDown (videoosd.js?v=10.5.0:1128)
```
</details>

_Just thoughts_
We could change `sendProgressUpdate` in `playbackmanager.js` to delay update by `volumechange`.
If there is no need to separate `volumechange` from other changes (i.e. accumulated update by `timeupdate`) we could clear this delay in `reportPlayback`.
If this will work then we probably can remove delays from volume sliders - playbackManager will do this job.

**Comment by MrTimscampi** [comment]
@mijofa @dmitrylyzo Any news on this?

**Comment by dmitrylyzo** [comment]
#1074 should fix spamming

File: src/controllers/playback/videoosd.js [hunk 1]
```diff
@@ -1158,6 +1178,29 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
...
+                case "1":
+                case "2":
+                case "3":
+                case "4":
+                case "5":
+                case "6":
+                case "7":
+                case "8":
+                case "9":
+                    var percent = parseInt(key);
```
Comment (MrTimscampi) [inline src/controllers/playback/videoosd.js:1200]: The radix for parseInt does **not** default to base 10. It should be explicitly set, even if it works as is.

See here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Description

**Comment by mijofa** [comment]
Sorry everyone, I've been a bit busy with a high workload and not kept up with this one properly.
Thanks for those changes, I'm quite happy with the code changes, although I'm a little uncomfortable with removing comments.
Comments never really hurt and help to describe why a perhaps silly looking piece of code shouldn't be removed. But if the standard for this project is less comments then fair enough, I'll probably still include a lot of comments in any future pull requests I may or may not make.

**Comment by dkanada** [comment]
> I'm quite happy with the code changes, although I'm a little uncomfortable with removing comments.

@mijofa noted, let us know in future pull requests that you'd like to leave the comments alone :) it looks like @MrTimscampi had made changes to get this merged but normally we would be happy to discuss changes with the original author.

**Pull request merged**